apps/gallery/test/marionette/delete_image_test.js

```diff
@@ -19,37 +19,117 @@ marionette('using the gallery menu', function() {
+    setup(function() {
+      // Remove all files in temp device storage.
+      client.fileManager.removeAllFiles();
+      // Add file into the pictures directory
+      client.fileManager.add({
+        type: 'pictures',
+        filePath: 'test_media/Pictures/firefoxOS.png'
+      });
+    });
+
+    suite('Select and delete single image', function() {
+      setup(function() {
+        app.launch();
+      });
+
+      test('should let you select images', function() {
+        // Use the 'selection' button and select the test image.
+        // Selected images should have an border/highlight.
+        app.thumbnailsSelectButton.click();
+        app.thumbnail.click();
+
+        var outline = app.thumbnail.cssProperty('outline');
+        assert.ok(outline != null);
+      });
 
-    var outline = app.thumbnail.cssProperty('outline');
-    assert.ok(outline != null);
+      test('should let you delete images', function() {
+        // This test deletes the test image.
+        app.thumbnailsSelectButton.click();
+        app.thumbnail.click();
+        app.thumbnailsDeleteButton.click();
+        app.confirmButton.click();
+
+        // Wait for the overlay with 'No photos or videos' message.
+        var overlayView = app.overlayView;
+        client.helper.waitForElement(overlayView);
+        assert.ok(overlayView.displayed());
+      });
+    });
   });
 
-  test('should let you delete images', function() {
-    // This test deletes the test image.
-    app.thumbnailsSelectButton.click();
-    app.thumbnail.click();
-    app.thumbnailsDeleteButton.click();
-    app.confirmButton.click();
-
-    // Wait for the overlay with 'No photos or videos' message.
-    var overlayView = app.overlayView;
-    client.helper.waitForElement(overlayView);
-    assert.ok(overlayView.displayed());
+  suite('Gallery app with multiple images', function() {
+    setup(function() {
+      // Remove all files in temp device storage.
+      client.fileManager.removeAllFiles();
+      // Add file into the pictures directory
+      client.fileManager.add([
+        {type: 'pictures',filePath: 'apps/gallery/test/images/01.jpg'},
+        {type: 'pictures',filePath: 'apps/gallery/test/images/02.png'},
+        {type: 'pictures',filePath: 'apps/gallery/test/images/03.gif'}
+      ]);
+    });
+
+    suite('Select and delete multiple images', function() {
+      setup(function() {
+        app.launch();
+      });
+
+      test('should let you select images', function() {
+        // Use the 'selection' button and select the test images.
+        // Selected images should have an border/highlight.
+        app.thumbnailsSelectButton.click();
+        app.tapThumbnail(0);
+        app.tapThumbnail(1);
+        app.tapThumbnail(2);
+
+        assert.ok(app.thumbnails[0].cssProperty('outline') != null);
+        assert.ok(app.thumbnails[1].cssProperty('outline') != null);
+        assert.ok(app.thumbnails[2].cssProperty('outline') != null);
```

Comment:
Consider adding an app.isThumbnailSelected(n) utility function so that you don't have brittle CSS dependencies like this directly in the test.